src/backend/renderer/gles2/mod.rs

```diff
@@ -709,6 +886,96 @@ impl ImportShm for Gles2Renderer {
     }
 }
 
+impl ImportMem for Gles2Renderer {
+    fn import_memory(
+        &mut self,
+        data: &[u8],
+        size: Size<i32, Buffer>,
+        flipped: bool,
+    ) -> Result<Gles2Texture, Gles2Error> {
+        self.make_current()?;
+
+        if data.len() < (size.w * size.h * 4) as usize {
+            return Err(Gles2Error::UnexpectedSize);
+        }
+
+        let texture = Gles2Texture(Rc::new({
+            let mut tex = 0;
+            unsafe {
+                self.gl.GenTextures(1, &mut tex);
+                self.gl.BindTexture(ffi::TEXTURE_2D, tex);
+                self.gl
+                    .TexParameteri(ffi::TEXTURE_2D, ffi::TEXTURE_WRAP_S, ffi::CLAMP_TO_EDGE as i32);
+                self.gl
+                    .TexParameteri(ffi::TEXTURE_2D, ffi::TEXTURE_WRAP_T, ffi::CLAMP_TO_EDGE as i32);
+                self.gl.TexImage2D(
+                    ffi::TEXTURE_2D,
+                    0,
+                    ffi::RGBA as i32,
+                    size.w,
+                    size.h,
+                    0,
+                    ffi::RGBA,
+                    ffi::UNSIGNED_BYTE as u32,
+                    data.as_ptr() as *const _,
+                );
+                self.gl.BindTexture(ffi::TEXTURE_2D, 0);
+            }
+            // new texture, upload in full
+            Gles2TextureInternal {
+                texture: tex,
+                texture_kind: 0,
+                is_external: false,
+                y_inverted: flipped,
+                size,
+                egl_images: None,
+                destruction_callback_sender: self.destruction_callback_sender.clone(),
+            }
+        }));
+
+        Ok(texture)
+    }
+
+    fn update_memory(
```

Comment:
It might be in the case of the `Gles2Renderer` (though it likely depends on the EGLImage/Dmabuf),
but the docs of the trait clearly state that:

```txt
    /// This function *may* error, if (but not limited to):
    /// - The texture was not created using either [`ImportMem::import_shm_buffer`] or [`ImportMem::import_memory`].
    /// ...
```